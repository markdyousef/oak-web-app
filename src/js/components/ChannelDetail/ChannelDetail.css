:root {
    --main-black-color: #000;
    --main-white-color: #fefeff;
    --main-orange-color: #ff9200;
    --main-red-color: #8b0000;
    --main-green-color: #005d6c;
}

.container {
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: column;
    background-color: #fff;
}

.container h2 {
    font-size: 18px;
    color: var(--main-white-color);
    padding: 20px;
}

.container nav {
    padding-left: 20px;
    display: flex;
    align-items: center;
    width: 100%;
    height: 60px;
    background-color: #fff;
    box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
    position: fixed;
    z-index: 999999;
}

.container nav a {
    font-size: 16px;
    color: var(--main-black-color);
    font-weight: bold;
}

.container nav h4 {
    margin-left: 5px;
    font-size: 20px;
    color: var(--main-green-color);
}

.membersContainer {
    padding-top: 60px;
    background-color: var(--main-green-color);
}

.membersContainer a {
    color: var(--main-white-color);
}
