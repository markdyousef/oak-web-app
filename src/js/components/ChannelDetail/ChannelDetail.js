// @flow
import React, { Component, PropTypes } from 'react';
import { Link } from 'react-router';

import css from './ChannelDetail.css';

import MessageStats from '../../containers/MessageStatsContainer';

class ChannelDetail extends Component {
    static propTypes = {
        data: PropTypes.shape({
            id: PropTypes.string,
            name: PropTypes.string,
            members: PropTypes.arrayOf(PropTypes.string),
            messages: PropTypes.arrayOf(PropTypes.object)
        }),
        getChannel: PropTypes.func.isRequired,
        params: PropTypes.shape({
            channelId: PropTypes.string
        }).isRequired,
        team: PropTypes.string.isRequired,
        isLoading: PropTypes.bool.isRequired
    };
    constructor() {
        super();
        this.renderMembers = this.renderMembers.bind(this);
        this.state = {};
    }
    componentWillMount() {
        const { team, params, getChannel } = this.props;
        getChannel(team, params.channelId);
    }
    renderMembers() {
        const { data } = this.props;

        // check if members is an array
        const members = (data.members) ? data.members : [];
        // some channels has no members
        if (members.length === 0) {
            return <div className={css.noMembers}>No members</div>;
        }
        return members.map(member => (
            <div className={css.member} key={member}>
                <div />
                <h4>{member}</h4>
            </div>
        ));
    }
    render() {
        const { data, isLoading } = this.props;

        if (isLoading) return <div>Loading</div>

        return (
            <div className={css.container}>
                <nav>
                    <Link to="/">CHANNELS /</Link>
                    <h1>{data.name}</h1>
                </nav>
                <div className={css.membersContainer}>
                    {this.renderMembers()}
                </div>
                <div className={css.messageContainer}>
                    <MessageStats channel={data.name} messages={data.messages} />
                </div>
            </div>
        );
    }
}

export default ChannelDetail;
