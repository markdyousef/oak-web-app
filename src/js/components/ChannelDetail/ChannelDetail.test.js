// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import ChannelDetail from './ChannelDetail';

test('renders ChannelDetail', () => {
    const props = {
        data: {
            id: 'id',
            name: 'general'
        }
    };
    const component = renderer.create(
        <ChannelDetail {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
