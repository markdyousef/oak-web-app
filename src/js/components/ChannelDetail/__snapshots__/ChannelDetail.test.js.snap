exports[`test renders ChannelDetail 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className={undefined}>
    <nav>
        <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/">
            CHANNELS /
        </Link>
        <h4>
            general
        </h4>
    </nav>
    <div
        className={undefined}>
        <h2>
            MEMBERS
        </h2>
        <Connect(TeamMembers) />
    </div>
    <div
        className={undefined}>
        <Connect(MessageStats)
            channel="general"
            messages={undefined} />
    </div>
    <div
        className={undefined}>
        <MessageTone
            tone={null}
            toneOther={Object {}} />
    </div>
</div>,
  "nodes": Array [
    <div
      className={undefined}>
      <nav>
            <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/">
                  CHANNELS /
            </Link>
            <h4>
                  general
            </h4>
      </nav>
      <div
            className={undefined}>
            <h2>
                  MEMBERS
            </h2>
            <Connect(TeamMembers) />
      </div>
      <div
            className={undefined}>
            <Connect(MessageStats)
                  channel="general"
                  messages={undefined} />
      </div>
      <div
            className={undefined}>
            <MessageTone
                  tone={null}
                  toneOther={Object {}} />
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ChannelDetail
        data={
                Object {
                        "id": "id",
                        "name": "general",
                      }
        }
        getChannel={[Function]}
        isLoading={false}
        params={
                Object {
                        "channelId": "id",
                      }
        }
        team="team name"
        tone={null} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ChannelDetail {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "data": Object {
            "id": "id",
            "name": "general",
          },
          "getChannel": [Function],
          "isLoading": false,
          "params": Object {
            "channelId": "id",
          },
          "team": "team name",
          "tone": null,
        },
        "refs": Object {},
        "state": Object {},
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className={undefined}>
          <nav>
                    <Link
                              onlyActiveOnIndex={false}
                              style={Object {}}
                              to="/">
                              CHANNELS /
                    </Link>
                    <h4>
                              general
                    </h4>
          </nav>
          <div
                    className={undefined}>
                    <h2>
                              MEMBERS
                    </h2>
                    <Connect(TeamMembers) />
          </div>
          <div
                    className={undefined}>
                    <Connect(MessageStats)
                              channel="general"
                              messages={undefined} />
          </div>
          <div
                    className={undefined}>
                    <MessageTone
                              tone={null}
                              toneOther={Object {}} />
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className={undefined}>
          <nav>
                    <Link
                              onlyActiveOnIndex={false}
                              style={Object {}}
                              to="/">
                              CHANNELS /
                    </Link>
                    <h4>
                              general
                    </h4>
          </nav>
          <div
                    className={undefined}>
                    <h2>
                              MEMBERS
                    </h2>
                    <Connect(TeamMembers) />
          </div>
          <div
                    className={undefined}>
                    <Connect(MessageStats)
                              channel="general"
                              messages={undefined} />
          </div>
          <div
                    className={undefined}>
                    <MessageTone
                              tone={null}
                              toneOther={Object {}} />
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ChannelDetail
    data={
        Object {
            "id": "id",
            "name": "general",
          }
    }
    getChannel={[Function]}
    isLoading={false}
    params={
        Object {
            "channelId": "id",
          }
    }
    team="team name"
    tone={null} />,
}
`;
